src/promptflow-azure/promptflow/azure/operations/_run_operations.py

```diff
@@ -732,6 +732,8 @@ def _resolve_flow_and_session_id(self, run: Run) -> Tuple[str, Optional[str]]:
         if run._use_remote_flow:
             return self._resolve_flow_definition_resource_id(run=run), None
         flow = load_flow(run.flow)
+        # set init kwargs for validation
+        flow._init_kwargs = run.init
```

Comment:
why we need to pass this to flow for validation? is it possible to valid the run directly?

Reply:
I have some concern about adding a member only with value in some specific scenarios

Reply:
Run does not aware flow type and signatures currently.

Reply:
Offline synced, currently will add init in cloud flow entity for 2 reasons:

1. ban model config with secrets in cloud run (since run entity does not have flow signatures)
2. prepare flow metadata in cloud run (requires init kwargs)

Reply:
Will add init for prepare_metadata in next PR